tools/transform.cpp

```diff
@@ -358,11 +358,21 @@ static void check_refinement(Errors &errs, Transform &t,
   };
 
   Solver::check({
-    { mk_fml(dom_a.notImplies(dom_b)),
+    { mk_fml(ub_a.notImplies(ub_b)),
       [&](const Result &r) {
         err(r, [](ostream&, const Model&){},
             "Source is more defined than target");
       }},
+    { mk_fml(ub_a && dom_a && !dom_b),
+      [&](const Result &r) {
+        err(r, [](ostream&, const Model&){},
+            "Target may never return");
+      }},
+    { mk_fml(ub_a && !dom_a && dom_b),
```

Comment:
well, let's compress these 2 queries in one to reduce overhead.

Reply:
Oh this wasn't intended. I thought I merged the two queries but it didn't. I'll update this